test/core/derivatives/implementation/autograd.py

```diff
@@ -40,17 +40,35 @@ def weight_jac_t_mat_prod(self, mat, sum_batch):
     def bias_jac_t_mat_prod(self, mat, sum_batch):
         return self.param_jac_t_mat_prod("bias", mat, sum_batch)
 
-    def param_jac_t_vec_prod(self, name, vec, sum_batch):
+    def bias_ih_l0_jac_t_mat_prod(self, mat, sum_batch):
+        return self.param_jac_t_mat_prod("bias_ih_l0", mat, sum_batch, axis_batch=1)
+
+    def param_jac_t_vec_prod(self, name, vec, sum_batch, axis_batch=0):
+        """Compute the product of jac_t and the given vector.
+
+        Args:
+            name (str): name of parameter for derivative
+            vec (torch.Tensor): vectors which to multiply
+            sum_batch (boolean): whether to sum along batch axis
+            axis_batch (int, optional): index of batch axis. Defaults to 0.
+
+        Returns:
+            torch.Tensor: product of jac_t and vec
+        """
         input, output, named_params = self.problem.forward_pass()
         param = named_params[name]
 
         if sum_batch:
             return transposed_jacobian_vector_product(output, param, vec)[0]
         else:
-            N = input.shape[0]
-
-            sample_outputs = [output[n] for n in range(N)]
-            sample_vecs = [vec[n] for n in range(N)]
+            sample_outputs = [
+                tensor.squeeze(dim=axis_batch)
+                for tensor in output.split(1, dim=axis_batch)
+            ]
+            sample_vecs = [
+                tensor.squeeze(dim=axis_batch)
```

Comment:
I know the old implementation squeezes out the batch dimension. However, I think it's safer to explicitly keep it and therefore effectively work with `batch_size=1`, as it avoids different behavior of layers on unbatched inputs. Can you remove the `squeeze`? The tests should still pass.

Reply:
Leads to a failed test.
When changing the line to `tensor` instead of `tensor.squeeze(dim=axis_batch)` the first test that fails in `make test-light` is `test/core/derivatives/derivatives_test.py::test_weight_jac_t_mat_prod[conv-example-dev=cpu-in=(3,2,7,7)-Conv2d(2,3,kernel_size=(2,2),stride=(2,2),padding=(1,1),dilation=(2,2),bias=False)-save_memory=True-sum_batch=False]` with error RuntimeError shape mismatch.

Reply:
Oh, so then there is a more general problem. Let's discuss in person.

Reply:
Note: Changing for sample_vecs and sample_outputs works.